File: hiking_gear_manager/lib/models/expedition.dart
// Similar to Kit, Expedition will have its own properties.
// The list of Kits and total weight calculation will be handled at a higher level.

class Expedition {
  int? id;
  String name;
  // List<Kit> kits; // Managed by service/UI layer

  Expedition({
    this.id,
    required this.name,
    // required this.kits, // Temporarily remove direct kit list here
  });

  // total_weight will be calculated dynamically.
  // double calculateTotalWeight(List<Kit> kits, /* function to get kit weight */)
  // This becomes more complex and depends on how Kit weights are resolved.
  // For now, let's assume this logic is in the service layer.

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
    };
  }

  static Expedition fromMap(Map<String, dynamic> map) {
    return Expedition(
      id: map['id'],
      name: map['name'],
      // kits are not directly stored in the Expedition table
    );
  }
}

File: app/models/expedition.py
from typing import List
from app.models.kit import Kit

class Expedition:
    def __init__(self, id: int, name: str, kits: List[Kit] = None):
        self.id = id
        self.name = name
        self.kits = kits if kits is not None else []

    def add_kit(self, kit: Kit):
        self.kits.append(kit)

    def remove_kit(self, kit: Kit):
        self.kits.remove(kit)

    @property
    def total_weight(self) -> float:
        return sum(kit.total_weight for kit in self.kits)
